core/src/core/transaction.rs

```diff
@@ -971,24 +982,40 @@ impl Transaction {
 /// and outputs.
 pub fn cut_through(inputs: &mut Vec<Input>, outputs: &mut Vec<Output>) -> Result<(), Error> {
 	// assemble output commitments set, checking they're all unique
-	let mut out_set = HashSet::new();
-	let all_uniq = { outputs.iter().all(|o| out_set.insert(o.commitment())) };
-	if !all_uniq {
+	outputs.sort_unstable_by(|v1, v2| v1.commitment().0.cmp(&v2.commitment().0));
+	if outputs
+		.windows(2)
+		.any(|pair| pair[0].commitment().0[..] == pair[1].commitment().0[..])
```

Comment:
While you are looking over this code - I wonder if there is a cleaner way of checking for uniqueness here.
Is there something in rust std lib that can do this for us?

Reply:
the only one off the top of my head is dedup and a length check, but that's a bit grosser IMO because it actually produces the dedup'd outputs array.

I'm using this as it is the paradigm used elsewhere in the code.